src/main/java/no/sikt/sws/Prefixer.java

```diff
@@ -0,0 +1,164 @@
+
+@JacocoGenerated
+public class Prefixer {
+
+    private static final Logger logger = LoggerFactory.getLogger(Prefixer.class);
+    public static final String INDEX = "_index";
+
+    /**
+     * Prefixing of URL.
+     * @param workspacePrefix prefix to apply
+     * @param resourceIdentifier url to prefix
+     * @return prefixed url
+     */
+    public static String url(String workspacePrefix, String resourceIdentifier) {
+        logger.debug("prefixing " + workspacePrefix + resourceIdentifier);
+        if (resourceIdentifier == null || resourceIdentifier.isEmpty()) {
+            return workspacePrefix + "-*";
+
+        } else if (resourceIdentifier.startsWith("_")) {
+            return resourceIdentifier;
+
+        }
+        return workspacePrefix + "-" + resourceIdentifier;
+    }
+
+    /**
+     * Prefixing of body.
+     * @param workspacePrefix prefix to apply
+     * @param resourceIdentifier url with index name
+     * @param gatewayBody body to prefix
+     * @return prefixed body
+     */
+    public static String body(String workspacePrefix, String resourceIdentifier, String gatewayBody)
+            throws BadRequestException {
+
+        if (gatewayBody == null) {
+            return null;
+        }
+
+        if (workspacePrefix == null) {
+            throw new IllegalArgumentException(REQUIRED_PARAMETER_IS_NULL + "workspacePrefix");
+        }
+
+        var searchCommand = OpenSearchCommand.fromString(resourceIdentifier);
+        logger.debug(searchCommand.name());
+        switch (searchCommand) {
+            case ALIAS:
+                return Prefixer.aliasBody(workspacePrefix,gatewayBody);
+            case BULK:
+                return Prefixer.bulkBody(workspacePrefix,getBulkBody(gatewayBody));
+            case SEARCH:
+                return Prefixer.searchBody(workspacePrefix,gatewayBody);
+            case MAPPING:
+                return Prefixer.mappingkBody(workspacePrefix,gatewayBody);
+            case DOC:
+                logger.debug("returning " + workspacePrefix + resourceIdentifier);
+                return gatewayBody;
+            case OTHER:
+                return Prefixer.indexesBody(workspacePrefix,gatewayBody);
+            case NOT_IMPLEMENTED:
+                throw new BadRequestException("Not implemented " + resourceIdentifier);
+            case INVALID:
+                throw new BadRequestException("resourceIdentifier [" + resourceIdentifier + "] is invalid");
+            default:
+                throw new IllegalStateException("Unexpected value: " + resourceIdentifier);
+        }
+    }
+
+
+    private static String mappingkBody(String workspacePrefix, String gatewayBody) {
+        logger.debug(workspacePrefix);
+        return gatewayBody;
+    }
+
+    private static String searchBody(String workspacePrefix, String gatewayBody) {
+        logger.debug(workspacePrefix);
+        return gatewayBody;
+    }
```

Comment:
for _doc you do the exact same as mappingkBody and searchBody but inside the switch.
Please do all of the "do-not-touch"-returns the same way: Either by calling the same function,  or share the same CASE in the switch